Lib/test/test_argparse.py

```diff
@@ -5716,6 +5716,31 @@ def test_double_dash(self):
         args = parser.parse_args(['--foo', 'a', 'b', '--', 'c', 'd'])
         self.assertEqual(NS(foo=['a', 'b'], bar=['c', 'd']), args)
 
+    def test_negative_number_success(self):
+        parser = argparse.ArgumentParser()
+        parser.add_argument('--int', type=int)
+        parser.add_argument('--float', type=float)
+
+        args = parser.parse_args(['--int', '-1000', '--float', '-1000.0'])
+        self.assertEqual(NS(int=-1000, float=-1000.0), args)
+
+        args = parser.parse_args(['--int', '-1_000', '--float', '-1_000.0'])
+        self.assertEqual(NS(int=-1000, float=-1000.0), args)
+
+        args = parser.parse_args(['--int', '-1_000_000', '--float', '-1_000_000.0'])
+        self.assertEqual(NS(int=-1000000, float=-1000000.0), args)
+
+    def test_negative_float_failure(self):
+        parser = ErrorRaisingArgumentParser(prog='PROG')
+        parser.add_argument('--float', type=float)
+
+        with self.assertRaises(ArgumentParserError) as cm:
+            parser.parse_args(['--float', '-_.45'])
```

Comment:
Each of these three statements should be in their own assertRaises + assertRegex, currently the second two don't run